modules/nextflow/src/main/resources/nextflow/executor/command-run.txt

```diff
@@ -107,6 +107,7 @@ on_exit() {
     [[ "$tee2" ]] && kill $tee2 2>/dev/null
     [[ "$ctmp" ]] && rm -rf $ctmp || true
     {{cleanup_cmd}}
+    sync || true
```

Comment:
I also do not see a problem to also use `sync` on network file system, it should always be supported. 

I think that same problem can happen on other platforms that are not Kubernetes and use overlay filesystems. Most likely we are not hitting this problem on other platforms because not all of them use the exit file to check the process exit status.

Reply:
and finally we got the `sync` crashing user pipelines!

Reply:
I also faced a non-terminating sync, but it turned out to be a problem with our filesystem stack.
Disabling the sync frequently led to missing exitcodes for me.

Reply:
Exit code error is better then hanging job, at least it can be retried!

Reply:
Currently, Nextflow does not handle the empty exitcode well, crashing the whole pipeline.

Reply:
Good point